File: Services/Models/Trip/DTO/TripDTO.cs
﻿using Models.Hotel.DTO;
using Models.Transport.DTO;

namespace Models.Trip.DTO
{
    public class TripDTO
    {
        public string HotelName { get; set; }

        public string Country { get; set; }

        public string City { get; set; }

        public DateOnly BeginDate { get; set; }

        public DateOnly EndDate { get; set; }

        public List<string> TypesOfMeals { get; set; }

        public List<RoomDTO> Rooms { get; set; }

        public FlightDTO ChosenFlight { get; set; }

        public List<FlightDTO> PossibleFlights { get; set; }

        public float Discount { get; set; }
        
        public int HotelId { get; set; }
    }
}

File: Frontend/frontend/src/app/dto/TripDTO.ts
import { FlightDTO } from './FlightDTO';
import { RoomDTO } from './RoomDTO';
export type TripDTO = {
  HotelName: string;
  Country: string;
  City: string;
  BeginDate: Date;
  EndDate: Date;
  TypesOfMeals: string[];
  Discount: number;
  RoomCombination?: number[];
  Rooms: RoomDTO[];
  ChosenFlight: FlightDTO;
  PossibleFlights: FlightDTO[];
  ChosenRooms?: RoomDTO[];
};
